Kernel: Using tolower() / toupper() from ctype.h in AK makes Kernel linking fail

I'm not sure why, it's explicitly included in the Kernel's `SOURCES`:

https://github.com/SerenityOS/serenity/blob/5729e76c7ddb755391e784dad7965fe65815fd57/Kernel/CMakeLists.txt#L261-L263

But using it in AK fails (`isfoo()` functions from `ctype.h` are used as well and work just fine):

```
[429/2788] Linking CXX executable Kernel/Kernel
FAILED: Kernel/Kernel 
: && /home/linus/Dev/git/serenity/Toolchain/Local/i686/bin/i686-pc-serenity-g++ -static -fsanitize=shift -fsanitize=shift-exponent -fsanitize=shift-base -fsanitize=integer-divide-by-zero -fsanitize=unreachable -fsanitize=vla-bound -fsanitize=return -fsanitize=signed-integer-overflow -fsanitize=bounds -fsanitize=bounds-strict -fsanitize=object-size -fsanitize=float-divide-by-zero -fsanitize=float-cast-overflow -fsanitize=nonnull-attribute -fsanitize=returns-nonnull-attribute -fsanitize=bool -fsanitize=enum -fsanitize=vptr -fsanitize=builtin -Wno-unknown-warning-option -pie -fPIE -fno-rtti -ffreestanding -fbuiltin -mno-80387 -mno-mmx -mno-sse -mno-sse2 -fno-asynchronous-unwind-tables -fstack-protector-strong -nostdlib -nostdinc -nostdinc++ -Wl,--hash-style=gnu,-z,relro,-z,now --sysroot /home/linus/Dev/git/serenity/Build/Root -Wl,-T /home/linus/Dev/git/serenity/Build/Kernel/linker.ld -nostdlib Kernel/CMakeFiles/Kernel.dir/ACPI/DynamicParser.cpp.o Kernel/CMakeFiles/Kernel.dir/ACPI/Initialize.cpp.o Kernel/CMakeFiles/Kernel.dir/ACPI/MultiProcessorParser.cpp.o Kernel/CMakeFiles/Kernel.dir/ACPI/Parser.cpp.o Kernel/CMakeFiles/Kernel.dir/AddressSanitizer.cpp.o Kernel/CMakeFiles/Kernel.dir/Arch/i386/CPU.cpp.o Kernel/CMakeFiles/Kernel.dir/Arch/i386/ProcessorInfo.cpp.o Kernel/CMakeFiles/Kernel.dir/Arch/i386/SafeMem.cpp.o Kernel/CMakeFiles/Kernel.dir/Arch/PC/BIOS.cpp.o Kernel/CMakeFiles/Kernel.dir/CMOS.cpp.o Kernel/CMakeFiles/Kernel.dir/CommandLine.cpp.o Kernel/CMakeFiles/Kernel.dir/Console.cpp.o Kernel/CMakeFiles/Kernel.dir/CoreDump.cpp.o Kernel/CMakeFiles/Kernel.dir/DMI.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/AsyncDeviceRequest.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/BXVGADevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/BlockDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/CharacterDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/Device.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/FullDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/I8042Controller.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/KeyboardDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/MBVGADevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/MemoryDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/NullDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/PCSpeaker.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/PS2MouseDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/RandomDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/SB16.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/SerialDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/USB/UHCIController.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/VMWareBackdoor.cpp.o Kernel/CMakeFiles/Kernel.dir/Devices/ZeroDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/Partition/DiskPartition.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/Partition/DiskPartitionMetadata.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/Partition/EBRPartitionTable.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/Partition/GUIDPartitionTable.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/Partition/MBRPartitionTable.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/Partition/PartitionTable.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/StorageDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/IDEController.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/IDEChannel.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/PATADiskDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/RamdiskController.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/RamdiskDevice.cpp.o Kernel/CMakeFiles/Kernel.dir/Storage/StorageManagement.cpp.o Kernel/CMakeFiles/Kernel.dir/DoubleBuffer.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/AnonymousFile.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/BlockBasedFileSystem.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/Custody.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/DevFS.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/DevPtsFS.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/Ext2FileSystem.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/FIFO.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/File.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/FileBackedFileSystem.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/FileDescription.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/FileSystem.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/Inode.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/InodeFile.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/InodeWatcher.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/Plan9FileSystem.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/ProcFS.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/TmpFS.cpp.o Kernel/CMakeFiles/Kernel.dir/FileSystem/VirtualFileSystem.cpp.o Kernel/CMakeFiles/Kernel.dir/FutexQueue.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/APIC.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/GenericInterruptHandler.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/IOAPIC.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/IRQHandler.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/InterruptManagement.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/PIC.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/SharedIRQHandler.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/SpuriousInterruptHandler.cpp.o Kernel/CMakeFiles/Kernel.dir/Interrupts/UnhandledInterruptHandler.cpp.o Kernel/CMakeFiles/Kernel.dir/KBufferBuilder.cpp.o Kernel/CMakeFiles/Kernel.dir/KSyms.cpp.o Kernel/CMakeFiles/Kernel.dir/Lock.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/E1000NetworkAdapter.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/IPv4Socket.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/LocalSocket.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/LoopbackAdapter.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/NE2000NetworkAdapter.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/NetworkAdapter.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/NetworkTask.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/RTL8139NetworkAdapter.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/Routing.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/Socket.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/TCPSocket.cpp.o Kernel/CMakeFiles/Kernel.dir/Net/UDPSocket.cpp.o Kernel/CMakeFiles/Kernel.dir/PCI/Access.cpp.o Kernel/CMakeFiles/Kernel.dir/PCI/Device.cpp.o Kernel/CMakeFiles/Kernel.dir/PCI/DeviceController.cpp.o Kernel/CMakeFiles/Kernel.dir/PCI/IOAccess.cpp.o Kernel/CMakeFiles/Kernel.dir/PCI/Initializer.cpp.o Kernel/CMakeFiles/Kernel.dir/PCI/MMIOAccess.cpp.o Kernel/CMakeFiles/Kernel.dir/Panic.cpp.o Kernel/CMakeFiles/Kernel.dir/PerformanceEventBuffer.cpp.o Kernel/CMakeFiles/Kernel.dir/Process.cpp.o Kernel/CMakeFiles/Kernel.dir/ProcessGroup.cpp.o Kernel/CMakeFiles/Kernel.dir/RTC.cpp.o Kernel/CMakeFiles/Kernel.dir/Random.cpp.o Kernel/CMakeFiles/Kernel.dir/Scheduler.cpp.o Kernel/CMakeFiles/Kernel.dir/StdLib.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscall.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/anon_create.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/abort.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/access.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/alarm.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/beep.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/chdir.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/chmod.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/chown.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/chroot.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/clock.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/debug.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/disown.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/dup2.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/execve.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/exit.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/fcntl.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/fork.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/ftruncate.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/futex.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/get_dir_entries.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/get_stack_bounds.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/getrandom.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/getuid.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/hostname.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/ioctl.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/keymap.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/kill.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/link.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/lseek.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/mkdir.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/mknod.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/mmap.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/module.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/mount.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/open.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/perf_event.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/pipe.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/pledge.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/prctl.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/process.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/profiling.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/ptrace.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/purge.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/read.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/readlink.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/realpath.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/rename.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/rmdir.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/sched.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/select.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/sendfd.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/setpgid.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/setuid.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/shutdown.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/sigaction.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/socket.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/stat.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/sync.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/sysconf.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/thread.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/times.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/ttyname.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/umask.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/uname.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/unlink.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/unveil.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/utime.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/waitid.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/watch_file.cpp.o Kernel/CMakeFiles/Kernel.dir/Syscalls/write.cpp.o Kernel/CMakeFiles/Kernel.dir/TTY/MasterPTY.cpp.o Kernel/CMakeFiles/Kernel.dir/TTY/PTYMultiplexer.cpp.o Kernel/CMakeFiles/Kernel.dir/TTY/SlavePTY.cpp.o Kernel/CMakeFiles/Kernel.dir/TTY/TTY.cpp.o Kernel/CMakeFiles/Kernel.dir/TTY/VirtualConsole.cpp.o Kernel/CMakeFiles/Kernel.dir/Tasks/FinalizerTask.cpp.o Kernel/CMakeFiles/Kernel.dir/Tasks/SyncTask.cpp.o Kernel/CMakeFiles/Kernel.dir/Thread.cpp.o Kernel/CMakeFiles/Kernel.dir/ThreadBlockers.cpp.o Kernel/CMakeFiles/Kernel.dir/ThreadTracer.cpp.o Kernel/CMakeFiles/Kernel.dir/Time/APICTimer.cpp.o Kernel/CMakeFiles/Kernel.dir/Time/HPET.cpp.o Kernel/CMakeFiles/Kernel.dir/Time/HPETComparator.cpp.o Kernel/CMakeFiles/Kernel.dir/Time/PIT.cpp.o Kernel/CMakeFiles/Kernel.dir/Time/RTC.cpp.o Kernel/CMakeFiles/Kernel.dir/Time/TimeManagement.cpp.o Kernel/CMakeFiles/Kernel.dir/TimerQueue.cpp.o Kernel/CMakeFiles/Kernel.dir/UBSanitizer.cpp.o Kernel/CMakeFiles/Kernel.dir/UserOrKernelBuffer.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/AnonymousVMObject.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/ContiguousVMObject.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/InodeVMObject.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/MemoryManager.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/PageDirectory.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/PhysicalPage.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/PhysicalRegion.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/PrivateInodeVMObject.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/ProcessPagingScope.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/PurgeablePageRanges.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/RangeAllocator.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/Region.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/SharedInodeVMObject.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/Space.cpp.o Kernel/CMakeFiles/Kernel.dir/VM/VMObject.cpp.o Kernel/CMakeFiles/Kernel.dir/WaitQueue.cpp.o Kernel/CMakeFiles/Kernel.dir/init.cpp.o Kernel/CMakeFiles/Kernel.dir/kprintf.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/ByteBuffer.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/FlyString.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/GenericLexer.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/Hex.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/JsonParser.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/JsonValue.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/LexicalPath.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/LogStream.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/String.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/StringBuilder.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/StringImpl.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/StringUtils.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/StringView.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/Time.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/Format.cpp.o Kernel/CMakeFiles/Kernel.dir/__/AK/UUID.cpp.o Kernel/CMakeFiles/Kernel.dir/__/Userland/Libraries/LibELF/Image.cpp.o Kernel/CMakeFiles/Kernel.dir/__/Userland/Libraries/LibELF/Validation.cpp.o Kernel/CMakeFiles/Kernel.dir/__/Userland/Libraries/LibVT/Terminal.cpp.o Kernel/CMakeFiles/Kernel.dir/__/Userland/Libraries/LibVT/Line.cpp.o Kernel/CMakeFiles/Kernel.dir/__/Userland/Libraries/LibKeyboard/CharacterMap.cpp.o Kernel/CMakeFiles/Kernel.dir/__/Userland/Libraries/LibCrypto/Cipher/AES.cpp.o Kernel/CMakeFiles/Kernel.dir/__/Userland/Libraries/LibCrypto/Hash/SHA2.cpp.o Kernel/CMakeFiles/Kernel.dir/__/Userland/Libraries/LibC/ctype.cpp.o -o Kernel/Kernel  Kernel/libkernel_heap.a  -lgcc  -lstdc++ && cd /home/linus/Dev/git/serenity/Build/Kernel && sh /home/linus/Dev/git/serenity/Kernel/mkmap.sh
/home/linus/Dev/git/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: Kernel/CMakeFiles/Kernel.dir/__/AK/StringUtils.cpp.o: in function `AK::StringUtils::to_snakecase(AK::StringView const&)':
./Build/../AK/StringUtils.cpp:364: undefined reference to `tolower(int)'
```

Note that this is not an issue with any current code, but I encountered it while adding something new - filing a bug so it can be fixed or I have something to reference when working around it. This probably *should* be working.

--- a/Userland/Libraries/LibC/stdarg.h
+++ b/Userland/Libraries/LibC/stdarg.h
@@ -26,19 +26,8 @@
 
 #pragma once
 
-#if defined(KERNEL)
-#    define __BEGIN_DECLS
-#    define __END_DECLS
-#else
-#    include <sys/cdefs.h>
-#endif
-
-__BEGIN_DECLS
-
 typedef __builtin_va_list va_list;
 
 #define va_start(v, l) __builtin_va_start(v, l)
 #define va_end(v) __builtin_va_end(v)
 #define va_arg(v, l) __builtin_va_arg(v, l)
-
-__END_DECLS

